subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/version/CommandLineVersionLocator.java

```diff
@@ -0,0 +1,169 @@
+        }
+
+        return installs;
+    }
+
+    private File findVswhereBinary() {
+        for (String programFilesKey : PROGRAM_FILES_KEYS) {
+            File programFilesDir;
+            try {
+                programFilesDir = new File(windowsRegistry.getStringValue(WindowsRegistry.Key.HKEY_LOCAL_MACHINE, REGISTRY_PATH_WINDOWS, programFilesKey));
+            } catch (MissingRegistryEntryException e) {
+                // We'll get this when we try to look up "ProgramFilesDir (x86)" on a 32-bit OS
+                continue;
+            }
+
+            File candidate = new File(programFilesDir, VISUAL_STUDIO_INSTALLER + "/" + VSWHERE_EXE);
+            if (candidate.exists() && candidate.isFile()) {
+                return candidate;
+            }
+        }
+
+        return os.findInPath(VSWHERE_EXE);
+    }
+
+    private String getVswhereOutput(File vswhereBinary, List<String> args) {
+        ExecAction exec = execActionFactory.newExecAction();
+        exec.args(args);
+        exec.executable(vswhereBinary.getAbsolutePath());
+        exec.setWorkingDir(vswhereBinary.getParentFile());
+
+        StreamByteBuffer buffer = new StreamByteBuffer();
+        exec.setStandardOutput(buffer.getOutputStream());
+        exec.setErrorOutput(NullOutputStream.INSTANCE);
+        exec.setIgnoreExitValue(true);
+        ExecResult result = exec.execute();
+
+        int exitValue = result.getExitValue();
+        if (exitValue == 0) {
+            return buffer.readAsString();
+        } else {
+            LOGGER.debug("vswhere.exe returned a non-zero exit value ({}) - ignoring", result.getExitValue());
+            return null;
+        }
+    }
+
+    private List<VisualStudioMetadata> parseJson(String json) {
+        List<VisualStudioMetadata> installs = Lists.newArrayList();
+        JsonReader reader = new JsonReader(new StringReader(json));
+        try {
+            reader.beginArray();
+            while (reader.hasNext()) {
+                installs.add(readInstall(reader));
+            }
+            reader.endArray();
+        } catch (IOException e) {
+            throw UncheckedException.throwAsUncheckedException(e);
+        }
+
+        return installs;
+    }
+
+    private VisualStudioMetadata readInstall(JsonReader reader) throws IOException {
+        String installPath = null;
+        String version = null;
+
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String key = reader.nextName();
+            if (key.equals(INSTALLATION_PATH_KEY)) {
+                installPath = reader.nextString();
+            } else if (key.equals(INSTALLATION_VERSION_KEY)) {
+                version = reader.nextString();
+            } else {
+                reader.skipValue();
+            }
+        }
+        reader.endObject();
+
+        File installDir = new File(installPath);
+        VisualCppMetadata visualCppMetadata = findVisualCppMetadata(installDir, version);
```

Comment:
which version is this?

Reply:
Visual studio version - I've renamed the variables to be more specific.